pkg/auto/wordpress/job/energy.go

```diff
@@ -0,0 +1,101 @@
+package job
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/vanti-dev/sc-bos/pkg/gen"
+)
+
+// EnergyJob gets energy consumed over the previous interval (typically 24 hours)
+type EnergyJob struct {
+	BaseJob
+	client     gen.MeterHistoryClient
+	infoClient gen.MeterInfoClient
+	Meters     []string
+	Interval   time.Duration
+}
+
+func (e *EnergyJob) GetName() string {
+	return "energy"
+}
+
+func (e *EnergyJob) GetClients() []any {
+	return []any{&e.client, &e.infoClient}
+}
+
+func (e *EnergyJob) GetInterval() time.Duration {
+	return e.Interval
+}
+
+func (e *EnergyJob) Do(ctx context.Context, sendFn sender) error {
+	if len(e.Meters) < 1 {
```

Comment:
Wouldn't it be better to do these checks for meters, sensors etc before you create the jobs and not start them if there are no devices?

Reply:
Ahh yes I did think about this. I thought the config could eventually be changed dynamically so didn't want this defined during the Smart-core bootup.

Reply:
That's a fair point but based on how the Job configs are set up, if you dynamically changed the url, ticker duration, endpoints etc. in the auto config you would have to restart the auto to capture those changes anyway, so I don't think it matters too much to restart to capture device changes either in this case